plugins/inputs/ipmi_sensor/ipmi.go

```diff
@@ -115,6 +125,30 @@ func (m *Ipmi) parse(acc telegraf.Accumulator, server string) error {
 		opts = conn.options()
 	}
 	opts = append(opts, "sdr")
+	if m.UseCache {
+		cacheFile := filepath.Join(m.CachePath, server + "_ipmi_cache")
+		_, err := os.Stat(cacheFile)
+		if os.IsNotExist(err) {
+			dumpOpts := opts
+			// init cache file
+			dumpOpts = append(dumpOpts, "dump")
+			dumpOpts = append(dumpOpts, cacheFile)
+			name := m.Path
+			if m.UseSudo {
+				// -n - avoid prompting the user for input of any kind
+				dumpOpts = append([]string{"-n", name}, dumpOpts...)
+				name = "sudo"
+			}
+			cmd := execCommand(name, dumpOpts...)
+			out, err := internal.CombinedOutputTimeout(cmd, m.Timeout.Duration)
+			if err != nil {
+				return fmt.Errorf("failed to run command %s: %s - %s", strings.Join(cmd.Args, " "), err, string(out))
+			}
+		}
+		// -S /tmp/cache_file speeds up ipmitool runs (by 2 to 10 times on HP servers)
```

Comment:
done with 7dab887
since I don't know the inner workings of ipmitool I can only do a informed guess about when you should not cache.
In most cases caching will help performance but nothing else however it may not hurt most people not using it.